File: Cards/Baron.lua
local ____lualib = require("lualib_bundle")
local __TS__Class = ____lualib.__TS__Class
local __TS__ClassExtends = ____lualib.__TS__ClassExtends
local ____exports = {}
local ____Enums = require("Enums")
local Ranks = ____Enums.Ranks
local ____Card = require("Cards.Card")
local Card = ____Card.default
____exports.default = __TS__Class()
local Baron = ____exports.default
Baron.name = "Baron"
__TS__ClassExtends(Baron, Card)
function Baron.prototype.____constructor(self, gameManager, suit)
    Card.prototype.____constructor(
        self,
        gameManager,
        suit,
        Ranks.BARON,
        2,
        5,
        "Baron"
    )
end
return ____exports

File: Cards/Baron.ts
import { Ranks, Suits } from "../Enums";
import GameManager from "../GameManager";
import Card from "./Card";

export default class Baron extends Card {
  constructor(gameManager: GameManager, suit: Suits) {
    super(
      gameManager,
      suit,
      Ranks.BARON,
      2,
      5,
      "Baron",
      "Assets/Images/BaronRank.png",
      {
        edelName: "Duke",
        edelPower: 9,
        edelValue: 5,
        edelRankAssetPath: "Assets/Images/DukeRank.png",
      }
    );
  }
}
